### PR TITLE
rework service

### PR DESCRIPTION
**Pull request by MariyaSin**

### CODE REVIEW

File: src/MessageService/Startup.cs [hunk 1]
```diff
@@ -21,228 +16,174 @@
...
+        .GetSection(BaseServiceInfoConfig.SectionName)
+        .Get<BaseServiceInfoConfig>();
 
-            Version = "1.3.1.0";
-            Description = "MessageService, is intended to work with the messages, emails and email templates.";
-            StartTime = DateTime.UtcNow;
-            ApiName = $"LT Digital Office - {_serviceInfoConfig.Name}";
-        }
+      _rabbitMqConfig = Configuration
+          .GetSection(BaseRabbitMqConfig.SectionName)
```
Comment (Cheephot) [inline src/MessageService/Startup.cs:96]: tab

**Review by Cheephot — APPROVED** [review]

File: src/MessageService.Broker/Consumers/CreateWorkspaceConsumer.cs [hunk 2]
```diff
@@ -14,7 +14,7 @@ public class CreateWorkspaceConsumer : IConsumer<ICreateWorkspaceRequest>
 
     private bool CreateWorkspace(ICreateWorkspaceRequest request)
     {
-      _workspaceRepository.Add(_workspaceMapper.Map(request.Name, request.Users, request.CreaterId));
+      _workspaceRepository.CreateAsync(_workspaceMapper.Map(request.Name, request.Users, request.CreaterId));
```
Comment (BelehovEgor) [inline src/MessageService.Broker/Consumers/CreateWorkspaceConsumer.cs:17]: await

File: src/MessageService.Business/Commands/Message/CreateMessageCommand.cs [hunk 3]
```diff
@@ -56,16 +55,15 @@ public class CreateMessageCommand : ICreateMessageCommand
...
-        return _responseCreator.CreateFailureResponse<Guid?>(HttpStatusCode.BadRequest);
-      }
-
       response.Status = OperationResultStatusType.FullSuccess;
 
       _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.Created;
 
+      if (response.Body is null)
+      {
+        response = _responseCreator.CreateFailureResponse<Guid?>(HttpStatusCode.BadRequest);
```
Comment (BelehovEgor) [inline src/MessageService.Business/Commands/Message/CreateMessageCommand.cs:64]: i guess here better return

File: src/MessageService.Data.Provider.MsSql.Ef/MessageServiceDbContext.cs [hunk 4]
```diff
@@ -16,16 +16,14 @@ public MessageServiceDbContext(DbContextOptions<MessageServiceDbContext> options
...
-    public DbSet<DbChannelUser> ChannelUsers { get; set; }
+    public DbSet<DbWorkspaceUser> WorkspacesUsers { get; set; }
     public DbSet<DbChannel> Channels { get; set; }
-    public DbSet<DbUnsentEmail> UnsentEmails { get; set; }
-    public DbSet<DbKeyword> ParseEntities { get; set; }
+    public DbSet<DbChannelUser> ChannelsUsers { get; set; }
+    public DbSet<DbMessage> Messages { get; set; }
+    public DbSet<DbThreadMessage> ThreadsMessages { get; set; }
+    public DbSet<DbMessageFile> MessagesFiles { get; set; }
+    public DbSet<DbMessageFile> MessagesImages { get; set; }
```
Comment (BelehovEgor) [inline src/MessageService.Data.Provider.MsSql.Ef/MessageServiceDbContext.cs:26]: are images file?

File: src/MessageService.Data.Provider.MsSql.Ef/Migrations/20211029184600_InitialTables.cs [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace LT.DigitalOffice.MessageService.Data.Provider.MsSql.Ef.Migrations
+{
+  [DbContext(typeof(MessageServiceDbContext))]
+  [Migration("20211029184600_InitialTables")]
+  public class InitialTables : Migration
+  {
+    protected override void Up(MigrationBuilder migrationBuilder)
```
Comment (BelehovEgor) [inline src/MessageService.Data.Provider.MsSql.Ef/Migrations/20211029184600_InitialTables.cs:12]: separate methods, why you don't drop prev tables, why don't you use prev tables?

File: src/MessageService.Models.Db/DbThreadMessage.cs [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    public Guid Id { get; set; }
+    public Guid MessageId { get; set; }
+    public string Content { get; set; }
+    public int Status { get; set; }
+    public Guid CreatedBy { get; set; }
+    public DateTime CreatedAtUtc { get; set; }
+    public Guid? ModifiedBy { get; set; }
+    public DateTime? ModifiedAtUtc { get; set; }
+
+    public DbMessage Message { get; set; }
```
Comment (BelehovEgor) [inline src/MessageService.Models.Db/DbThreadMessage.cs:20]: now without links to files or images?

File: src/MessageService.Models.Dto/MessageService.Models.Dto.csproj [hunk 7]
```diff
@@ -2,4 +2,8 @@
 
   <Import Project="$(ProjectDir)..\..\Common.Configurations.targets" />
 
+  <ItemGroup>
+    <Folder Include="Responses\" />
```
Comment (BelehovEgor) [inline src/MessageService.Models.Dto/MessageService.Models.Dto.csproj:6]: why not the usual folder?

File: src/MessageService/Startup.cs [hunk 8]
```diff
@@ -21,228 +16,159 @@
...
 
-        private void StartResender(IApplicationBuilder app)
+        x.UsingRabbitMq((context, cfg) =>
         {
-            EmailEngineConfig emailEngineConfig = Configuration
-                .GetSection(EmailEngineConfig.SectionName)
-                .Get<EmailEngineConfig>();
+          cfg.Host(_rabbitMqConfig.Host, "/", host =>
+          {
+            host.Username($"{_serviceInfoConfig.Name}_{_serviceInfoConfig.Id}");
```
Comment (BelehovEgor) [inline src/MessageService/Startup.cs:43]: make it like auth/ you should get data from env variable or appsettings

File: test/MessageService.Data.UnitTests/ChannelRepositoryTests.cs [hunk 9]
```diff
@@ -46,7 +46,7 @@ public void ShouldAddSuccesfuly()
         {
             DbChannel channel = new DbChannel();
 
-            _repository.Add(channel);
+            _repository.CreateAsync(channel);
```
Comment (BelehovEgor) [inline test/MessageService.Data.UnitTests/ChannelRepositoryTests.cs:49]: await

File: src/MessageService.Data.Provider.MsSql.Ef/Migrations/20211029184600_InitialTables.cs (hunk 5 above)
Comment (MariyaSin) [reply src/MessageService.Data.Provider.MsSql.Ef/Migrations/20211029184600_InitialTables.cs:12]: now this service does not store valuable information, we could clean up migrations

File: test/MessageService.Data.UnitTests/ChannelRepositoryTests.cs (hunk 9 above)
Comment (MariyaSin) [reply test/MessageService.Data.UnitTests/ChannelRepositoryTests.cs:49]: removed test

File: src/MessageService.Models.Dto/MessageService.Models.Dto.csproj (hunk 7 above)
Comment (MariyaSin) [reply src/MessageService.Models.Dto/MessageService.Models.Dto.csproj:6]: reworked

**Review by VLiamin — APPROVED** [review]

**Review by ndoriomedov — APPROVED** [review]

File: src/MessageService/Startup.cs [hunk 10]
```diff
@@ -21,228 +16,159 @@
...
-            IServiceProvider serviceProvider = app.ApplicationServices.GetRequiredService<IServiceProvider>();
+      _rabbitMqConfig = Configuration
+        .GetSection(BaseRabbitMqConfig.SectionName)
+        .Get<RabbitMqConfig>();
 
-            foreach(KeyValuePair<string, string> pair in _rabbitMqConfig.FindUserParseEntitiesEndpoint)
-            {
-                IRequestClient<IFindParseEntitiesRequest> rcFindParseEntities = serviceProvider.CreateRequestClient<IFindParseEntitiesRequest>(
-                    new Uri($"{_rabbitMqConfig.BaseUrl}/{pair.Value}"), default);
+      Version = "1.3.1.0";
```
Comment (Cheephot) [inline src/MessageService/Startup.cs:86]: 1.3.1.1

File: src/MessageService/Startup.cs (hunk 8 above)
Comment (MariyaSin) [reply src/MessageService/Startup.cs:43]: I will do it in the next PR, it seems to me that the GetRabbitMqCredentials method can be transferred to the kernel

File: src/MessageService.Business/Commands/Message/CreateMessageCommand.cs (hunk 3 above)
Comment (MariyaSin) [reply src/MessageService.Business/Commands/Message/CreateMessageCommand.cs:64]: return will be in next line

**Pull request merged**